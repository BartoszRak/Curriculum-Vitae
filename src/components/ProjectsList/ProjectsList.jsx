import React from 'react'
import PropTypes from 'prop-types'

import ProjectTile from './ProjectTile'

import useStyle from './ProjectsList.style'

export function ProjectsList({ classes: overridingClasses, data }) {
  const classes = { ...useStyle(), ...overridingClasses }
  return (
    <div className={classes.root}>
      {data.map(project => (
        <ProjectTile key={project.title} {...project} />
      ))}
    </div>
  )
}

ProjectsList.propTypes = {
  classes: PropTypes.object,
  data: PropTypes.arrayOf(
    PropTypes.shape({
      description: PropTypes.string,
      isFavorite: PropTypes.bool,
      tags: PropTypes.arrayOf(
        PropTypes.shape({
          name: PropTypes.string,
          url: PropTypes.string,
        })
      ),
      title: PropTypes.string.isRequired,
      url: PropTypes.string,
      warning: PropTypes.string,
    })
  ),
}

export default ProjectsList
