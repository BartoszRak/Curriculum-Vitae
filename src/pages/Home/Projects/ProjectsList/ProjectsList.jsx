import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core'

import ProjectTile from './ProjectTile'
import technologiesMap from '~utils/technologiesMaps'

import style from './ProjectsList.style'

const projects = [
  {
    description:
      'Carefully designed react boilerplate repositorium that gives you fast and smooth start! ',
    name: 'React Awesome Boilerplate',
    tags: [technologiesMap.react, technologiesMap.redux, technologiesMap.rematch],
    url: 'https://github.com/BartoszRak/React-Awesome-Boilerplate',
  },
  {
    description: 'Matrices computing library.',
    name: 'Mx',
    tags: [technologiesMap.typescript],
    url: 'https://github.com/BartoszRak/Mx',
  },
]

export function ProjectsList({ classes }) {
  return (
    <div className={classes.root}>
      {projects.map(project => {
        const {
          description, name, url, tags,
        } = project
        return (
          <ProjectTile
            description={description}
            key={name}
            name={name}
            tags={tags}
            url={url}
          />
        )
      })}
    </div>
  )
}

ProjectsList.propTypes = {
  classes: PropTypes.object,
}

export default withStyles(style)(ProjectsList)
