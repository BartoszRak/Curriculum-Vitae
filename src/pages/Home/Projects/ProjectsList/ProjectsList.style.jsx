import { makeStyles } from '@material-ui/core'

export default makeStyles(({ breakpoints }) => ({
  root: {
    columnGap: 40,
    display: 'block',
    marginTop: 40,
    [breakpoints.up('lg')]: {
      columnCount: 4,
    },
    [breakpoints.down('md')]: {
      columnCount: 3,
    },
    [breakpoints.down('sm')]: {
      columnCount: 2,
    },
    [breakpoints.down('sm')]: {
      columnCount: 1,
    },
  },
}))
