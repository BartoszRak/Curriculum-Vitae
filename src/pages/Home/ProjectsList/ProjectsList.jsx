import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core'

import ProjectTile from './ProjectTile'

import style from './ProjectsList.style'

import ForestImage from '../../../assets/images/projects/forest-sunny-green.jpg'
import CoastImage from '../../../assets/images/projects/coast-sunny.jpg'

export function ProjectsList({ classes }) {
  const projects = [
    {
      image: ForestImage,
      name: 'React Awesome Boilerplate',
      description: 'Carefully designed react boilerplate repositorium that gives you fast and smooth start!',
      url: 'https://github.com/BartoszRak/React-Awesome-Boilerplate',
    }, {
      image: CoastImage,
      name: 'Mx',
      description: 'Matrices computing library.',
      url: 'https://github.com/BartoszRak/Mx',
    }, 
  ]

  return (
    <div className={classes.root}>
      {projects.map(project => {
        const { description, name, url, image } = project
        return (
          <div className={classes.cardHolder} key={name}>
            <ProjectTile
              image={image}
              name={name}
              description={description}
              url={url}
            />
          </div>
        )
      })}
    </div>
  )
}

ProjectsList.propTypes = {
  classes: PropTypes.object,
}

export default withStyles(style)(ProjectsList)
