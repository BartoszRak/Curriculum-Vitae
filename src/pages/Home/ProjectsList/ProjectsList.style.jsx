export default ({ spacing }) => ({
  cardHolder: {
    alignItems: 'stretch',
    border: '1px solid red',
    display: 'flex',
    height: 300,
    justifyContent: 'center',
    padding: spacing.unit * 2,
    width: 300,
  },
  root: {
    border: '1px solid red',
    display: 'flex',
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'center',
  },
})
