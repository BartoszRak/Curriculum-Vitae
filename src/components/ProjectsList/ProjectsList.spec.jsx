import React from 'react'
import { cleanup } from '@testing-library/react'
import { shallow } from 'enzyme'

import { ProjectsList } from './ProjectsList'

jest.unmock('@material-ui/core')
jest.unmock('@material-ui/icons')

jest.mock('./ProjectTile', () => 'ProjectTileMock')

let data

describe('component: ProjectsList', () => {
  afterEach(cleanup)

  beforeEach(() => {
    data = [{
      description: '1 tile test description',
      isFavorite: false,
      tags: [],
      title: '1 tile test title',
      url: 'http://1tiletesturl.com',
      warning: 'sample warning',
    }, {
      description: '2 tile test description',
      isFavorite: false,
      tags: [],
      title: '2 tile test title',
      url: 'http://2tiletesturl.com',
    }]
  })

  describe('rendering', () => {
    test('render without crush', () => {
      const wrapper = shallow(<ProjectsList data={data} />)

      expect(wrapper).toBeTruthy()
    })

    test('match snapshot ', () => {
      const wrapper = shallow(<ProjectsList data={data} />)

      expect(wrapper).toMatchSnapshot()
    })

    test('render valid number of tiles ', () => {
      const wrapper = shallow(<ProjectsList data={data} />)

      expect(wrapper.find('ProjectTileMock').length).toBe(data.length)
    })
  })
})
