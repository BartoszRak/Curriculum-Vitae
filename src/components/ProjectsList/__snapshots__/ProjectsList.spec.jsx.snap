// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: ProjectsList rendering match snapshot  1`] = `
<div
  className="makeStyles-root-1"
>
  <ProjectTileMock
    key="1 tile test title"
    project={
      Object {
        "description": "1 tile test description",
        "isFavorite": false,
        "tags": Array [],
        "title": "1 tile test title",
        "url": "http://1tiletesturl.com",
      }
    }
  />
  <ProjectTileMock
    key="2 tile test title"
    project={
      Object {
        "description": "2 tile test description",
        "isFavorite": false,
        "tags": Array [],
        "title": "2 tile test title",
        "url": "http://2tiletesturl.com",
      }
    }
  />
</div>
`;
